include/pybind11/numpy.h

```diff
@@ -244,7 +244,8 @@ template <typename T> struct is_complex : std::false_type { };
 template <typename T> struct is_complex<std::complex<T>> : std::true_type { };
 
 template <typename T> using is_pod_struct = all_of<
-    std::is_pod<T>, // since we're accessing directly in memory we need a POD type
+    std::is_standard_layout<T>,     // since we're accessing directly in memory we need a standard layout type
+    std::is_trivially_copyable<T>,
```

Comment:
Closer still:
```
std::is_trivially_destructible<T>, satisifies_any_of<T, std::has_trivial_copy_constructor, std::has_trivial_copy_assign>
```
libstdc++ doesn't appear to have an equivalent `has_trivial_move_constructor`, but the above should handle most cases.